### PR TITLE
Enable tracking dataset support

### PR DESCRIPTION
**Pull request by challen-zhou**
1. Add functions to enable dataset support in tracking module.
2. Add one regression tool to track tracking precision and perf.

### CODE REVIEW

**Comment by challen-zhou** [comment]
@sharronliu @yechun1 , please help review the change.

**Comment by challen-zhou** [comment]
@yechun1 , could you please check and review one more change?

File: object_analytics_node/CMakeLists.txt [hunk 1]
```diff
@@ -195,6 +214,11 @@ install(TARGETS
   DESTINATION lib/${PROJECT_NAME}
 )
 
+install(TARGETS
```
Comment (yechun1) [inline object_analytics_node/CMakeLists.txt]: would you please notice the example usage of the tool somewhere? in readme or usage code?

File: object_analytics_node/src/dataset/tr_dataset.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <string>
+#include <vector>
+
+#include "object_analytics_node/dataset/track_dataset.hpp"
```
Comment (yechun1) [inline object_analytics_node/src/dataset/tr_dataset.cpp]: this dataset feature is only for tracker or could be used for others? if only for tracker, I would suggest put .hpp to `object_analytics_node/tracker/track_dataset.hpp`. Save suggests with cpp file to put under tracker folder.

**Comment by challen-zhou** [comment]
> @challen-zhou thanks for the PR. See you have modify the code style, not sure which tool you have used. would you please check with`ament build -j8 --symlink-install --isolated src/ros2_object_analytics`

**Comment by challen-zhou** [comment]
> > @challen-zhou thanks for the PR. See you have modify the code style, not sure which tool you have used. would you please check with`ament build -j8 --symlink-install --isolated src/ros2_object_analytics`

The patches already passed "ament build"

File: object_analytics_node/src/dataset/tr_dataset.cpp (hunk 2 above)
Comment (challen-zhou) [reply object_analytics_node/src/dataset/tr_dataset.cpp]: > this dataset feature is only for tracker or could be used for others? if only for tracker, I would suggest put .hpp to `object_analytics_node/tracker/track_dataset.hpp`. Save suggests with cpp file to put under tracker folder.

Datasets is a standalone feature which would suppose to be in separated folder, the folder should contain segementation/location dataset later.

File: object_analytics_node/CMakeLists.txt (hunk 1 above)
Comment (challen-zhou) [reply object_analytics_node/CMakeLists.txt]: This will be addressed in readme once the code got merged.

**Comment by challen-zhou** [comment]
@yechun1 , new commits added, please help check. Also please note that the coding style check passed.

**Comment by yechun1** [comment]
@challen-zhou  Would you please help to resubmit commits or merge patches to remove those unnecessary code style changes between previous commits and last commits?

**Pull request merged**